common/persistence/nosql/nosql_task_store_test.go

```diff
@@ -104,3 +109,175 @@ func TestGetTaskListSize(t *testing.T) {
 		size,
 	)
 }
+
+func getValidLeaseTaskListRequest() *persistence.LeaseTaskListRequest {
+	return &persistence.LeaseTaskListRequest{
+		DomainID:     TestDomainID,
+		DomainName:   TestDomainName,
+		TaskList:     TestTaskListName,
+		TaskType:     int(types.TaskListTypeDecision),
+		TaskListKind: int(types.TaskListKindNormal),
+		RangeID:      0,
+	}
+}
+
+func getExpectedTaskListRow() *nosqlplugin.TaskListRow {
+	return &nosqlplugin.TaskListRow{
+		DomainID:        TestDomainID,
+		TaskListName:    TestTaskListName,
+		TaskListType:    int(types.TaskListTypeDecision),
+		RangeID:         initialRangeID,
+		TaskListKind:    int(types.TaskListKindNormal),
+		AckLevel:        initialAckLevel,
+		LastUpdatedTime: time.Now(),
+	}
+}
+
+func checkTaskListRowExpected(t *testing.T, taskList *nosqlplugin.TaskListRow) {
+	expectedRow := getExpectedTaskListRow()
+	// Check the duration
+	assert.WithinDuration(t, expectedRow.LastUpdatedTime, taskList.LastUpdatedTime, time.Second)
+
+	// Set the expected time to the actual time to make the comparison work
+	expectedRow.LastUpdatedTime = taskList.LastUpdatedTime
+	assert.Equal(t, expectedRow, taskList)
+}
+
+func checkTaskListInfoExpected(t *testing.T, taskListInfo *persistence.TaskListInfo) {
+	assert.Equal(t, TestDomainID, taskListInfo.DomainID)
+	assert.Equal(t, TestTaskListName, taskListInfo.Name)
+	assert.Equal(t, int(types.TaskListTypeDecision), taskListInfo.TaskType)
+	assert.Equal(t, initialRangeID, taskListInfo.RangeID)
+	assert.Equal(t, initialAckLevel, taskListInfo.AckLevel)
+	assert.Equal(t, int(types.TaskListKindNormal), taskListInfo.Kind)
+	assert.WithinDuration(t, time.Now(), taskListInfo.LastUpdated, time.Second)
+}
+
+func TestLeaseTaskList_emptyTaskList(t *testing.T) {
+	store, _ := setupNoSQLStoreMocks(t)
+
+	req := getValidLeaseTaskListRequest()
+	req.TaskList = ""
+	_, err := store.LeaseTaskList(ctx.Background(), req)
+
+	var expectedErr *types.InternalServiceError
+	assert.ErrorAs(t, err, &expectedErr)
+	assert.ErrorContains(t, err, "LeaseTaskList requires non empty task list")
+}
+
+func TestLeaseTaskList_selectErr(t *testing.T) {
+	store, db := setupNoSQLStoreMocks(t)
+
+	req := getValidLeaseTaskListRequest()
+	db.EXPECT().SelectTaskList(gomock.Any(), getExpectedTaskListFilter()).Return(nil, assert.AnError)
+	// The error is _not_ a NotFoundError
+	db.EXPECT().IsNotFoundError(assert.AnError).Return(false).Times(2)
+	db.EXPECT().IsTimeoutError(assert.AnError).Return(true).Times(1)
+
+	_, err := store.LeaseTaskList(ctx.Background(), req)
+
+	assert.Error(t, err)
+	// The function does not wrap the error, it just adds it to the message
+	assert.ErrorContains(t, err, assert.AnError.Error())
+}
+
+func TestLeaseTaskList_selectErrNotFound(t *testing.T) {
```

Comment:
I think it would be easier to read this as a table test:
```
for _, tc := []struct{
   name string
   prepareDB(*db.Mock)
   assertError func(t *testing.T, err error)
   assertResult func(t * testing.T, res)
}
```
It would be easier to assess if all possible cases are covered, and it should be easier to read.